### PR TITLE
Update client maps to be sync map

### PR DESCRIPTION
**Pull request by kneal**

### CODE REVIEW

File: executor/linux/service.go [hunk 1]
```diff
@@ -81,7 +82,12 @@ func (c *client) PlanService(ctx context.Context, ctn *pipeline.Container) error
 func (c *client) ExecService(ctx context.Context, ctn *pipeline.Container) error {
 	b := c.build
 	r := c.repo
-	l := c.serviceLogs[ctn.ID]
+
+	result, ok := c.serviceLogs.Load(ctn.ID)
+	if !ok {
+		return fmt.Errorf("unable to get log from client")
```
Comment (jbrockopp) [inline executor/linux/service.go]: Thoughts on updating the error message for the resource here?

Maybe something like:

`unable to get service log from client`

File: executor/linux/step.go [hunk 2]
```diff
@@ -139,7 +139,12 @@ func (c *client) PlanStep(ctx context.Context, ctn *pipeline.Container) error {
 func (c *client) ExecStep(ctx context.Context, ctn *pipeline.Container) error {
 	b := c.build
 	r := c.repo
-	l := c.stepLogs[ctn.ID]
+
+	result, ok := c.stepLogs.Load(ctn.ID)
+	if !ok {
+		return fmt.Errorf("unable to get log from client")
```
Comment (jbrockopp) [inline executor/linux/step.go]: Potentially something similar to above?

i.e. `unable to get step log from client`

File: executor/linux/service.go (hunk 1 above)
Comment (kneal) [reply executor/linux/service.go]: Yeah being a bit more verbose would definitely help the parsing when you're looking at the output in a log aggregator. Will update!

File: executor/linux/step.go (hunk 2 above)
Comment (kneal) [reply executor/linux/step.go]: Same as above. 😄

**Pull request merged**